src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListNotesModal.tsx

```diff
@@ -0,0 +1,271 @@
+
+export interface FormInput {
+  taskId: String;
+  title: String;
+  notes: String;
+  assignees: number[];
+  deadline: Date;
+  priority: TaskPriority;
+}
+
+const schema = yup.object().shape({
+  notes: yup.string().required(),
+  deadline: yup.date().required(),
+  priority: yup.string().required(),
+  assignees: yup.array().required(),
+  title: yup.string().required()
+});
+
+const TaskListNotesModal: React.FC<TaskListNotesModalProps> = ({
+  task,
+  modalShow,
+  onHide,
+  onSubmit
+}: TaskListNotesModalProps) => {
+  const { isLoading, isError, error, data: users } = useAllUsers();
+  const auth = useAuth();
+  const [isEditMode, setIsEditMode] = useState(false);
+  const {
+    handleSubmit,
+    control,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      title: task.title,
+      taskId: task.taskId,
+      notes: task.notes,
+      deadline: task.deadline,
+      priority: task.priority,
+      assignees: task.assignees.map((assignee) => assignee.userId)
+    }
+  });
+  if (!auth.user || isLoading || !users) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const userToAutocompleteOption = (user: User): { label: string; id: number } => {
+    return { label: `${fullNamePipe(user)} (${user.email})`, id: user.userId };
+  };
+  const options = users.sort((a, b) => (a.firstName > b.firstName ? 1 : -1)).map(userToAutocompleteOption);
+  const dialogWidth: Breakpoint = 'md';
+  const renderModal: () => JSX.Element = () => {
+    return (
+      <Dialog fullWidth maxWidth={dialogWidth} open={modalShow} onClose={onHide}>
+        <DialogTitle className={'font-weight-bold'}>{task.title}</DialogTitle>
+        <DialogContent>
+          <Grid container spacing={2}>
+            <Grid item xs={12} md={6}>
+              <Typography fontWeight={'bold'}>
+                Priority: <Typography display={'inline'}> {task.priority}</Typography>
+              </Typography>
+            </Grid>
+            <Grid item xs={11} md={5}>
+              <Typography fontWeight={'bold'}>
+                Deadline:
+                <Typography display={'inline'}> {datePipe(new Date(task.deadline))}</Typography>
+              </Typography>
+            </Grid>
+            <Grid item xs={1} md={1}>
+              <IconButton onClick={() => setIsEditMode(true)}>
+                <Edit />
+              </IconButton>
+            </Grid>
+            <Grid item xs={12} md={8}>
+              <Typography fontWeight={'bold'}>
+                Assignees(s):
+                <Typography display={'inline'}> {task.assignees.map((user) => fullNamePipe(user)).join(', ')}</Typography>
+              </Typography>
+            </Grid>
+            <Grid item xs={12} md={12}>
+              <Typography fontWeight={'bold'}>Notes:</Typography>
+              <Box sx={{ height: '200px', overflow: 'auto', border: 'solid', borderWidth: '1', borderColor: '#EF4345' }}>
```

Comment:
I don't think we need this outlined

Reply:
We should test this with a very longboi notes. I'm not sure how this feature is gonna get used but if the notes are very long we want to make sure it looks good (like if we need to include a scrollbar or something)

Reply:
Notes are capped at 250